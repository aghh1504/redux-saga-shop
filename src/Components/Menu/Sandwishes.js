/*eslint-disable no-unused-vars */
import React, { Component } from 'react'

export default class Sandwishes extends Component {
  render() {
    const {sandwishes, checkedInput} = this.props
      return (
          <div>
            {sandwishes && sandwishes.map((item, i) => (
                  <div key={i}>
                    <h4 style={{display: 'inline-block'}}>{item.name}</h4>
                    <input type='checkbox' id={item.id} style={{display: 'inline-block'}} checked={item.isChecked} onChange={() => checkedInput(item.id)}/>
                  </div>
            ))}
          </div>
      )
    }
}
