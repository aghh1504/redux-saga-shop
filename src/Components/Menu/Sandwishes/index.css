.sandwishes_name {
  display: inline-block;
}

.sandwishes_price {
  display: inline-block;
  margin: 0 0.8rem;
}

.sandwishes_checkbox {
  display: inline-block;
}
.sandwishes_extra {
  display: inline-block;
}
