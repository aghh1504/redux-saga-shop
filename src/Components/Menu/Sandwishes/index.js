/*eslint-disable no-unused-vars */
import React, { Component } from 'react'
import './index.css'

export default class Sandwishes extends Component {
  state={
    quantity:0
  }

    onHandleQuantity = (e) => {
      this.setState({quantity: e.target.value})
    }
  render() {
    const {sandwishes, addToBasket} = this.props
      return (
          <div>
            {sandwishes && sandwishes.map(item => (
                  <div key={item.id}>
                    <h4 className='sandwishes_name'>{item.name}</h4>
                    <h5 className='sandwishes_price'>Price ${item.price}</h5>
                    <input  type='number' placeholder="quantity" onChange={this.onHandleQuantity}/>
                    <button id={item.id} className='sandwishes_checkbox' onClick={() => addToBasket(item, this.state.quantity)}>Add to Basket</button>
                    {item.options && item.options.map((item,i) => (
                      <div className='sandwishes_extra' key={i}>
                        <div className='sandwishes_extra'>Add Extra: {item.name} - ${item.price}</div>
                        <button id={item.id} className='sandwishes_checkbox' onClick={() => addToBasket(item)}>Add to Basket</button>
                      </div>
                    ))}
                  </div>
            ))}
          </div>
      )
    }
}
