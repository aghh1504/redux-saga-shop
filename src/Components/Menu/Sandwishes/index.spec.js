import React from 'react';
import ReactDOM from 'react-dom';
import Sandwishes from './index';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Component: Sandwishes', () => {
  const sandwishes = [
      {
          "id": 1,
          "preparationTimeSeconds": 600,
          "name": "Tikka Naan Roll",
          "description": "Filets de poulet marinés et rôtis au Tandoor roulé dans un naan",
          "options": [
              {
                  "name": "Supplement fromage",
                  "price": 1
              }
          ],
          "price": 5
      },
      {
          "id": 2,
          "preparationTimeSeconds": 600,
          "name": "Sheek Naan Roll",
          "description": "Brochette de boeuf haché rôtie au Tandoor roulée dans un naan",
          "options": [
              {
                  "name": "Supplement fromage",
                  "price": 1
              }
          ],
          "price": 5
      },
      {
          "id": 3,
          "preparationTimeSeconds": 600,
          "name": "Pakora Naan Roll",
          "description": "Beignets d'aubergines à la farine de poix chiche roulés dans un naan",
          "options": [
              {
                  "name": "Supplement fromage",
                  "price": 1
              }
          ],
          "price": 5
      },
      {
          "id": 4,
          "preparationTimeSeconds": 600,
          "name": "Paneer Naan Roll",
          "description": "Fromage indien maison mariné et rôti au Tandoor roulé dans un naan",
          "options": [
              {
                  "name": "Supplement fromage",
                  "price": 1
              }
          ],
          "price": 6.50
      },
      {
          "id": 5,
          "preparationTimeSeconds": 300,
          "name": "Frites",
          "price": 2.50
      },
      {
          "id": 6,
          "preparationTimeSeconds": 300,
          "name": "Pakoras",
          "price": 3
      }
  ]

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Sandwishes addToBasket={jest.fn()} sandwishes={sandwishes}/>
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
