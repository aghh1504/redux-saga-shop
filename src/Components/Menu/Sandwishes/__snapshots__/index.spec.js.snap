// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Sandwishes should match its empty snapshot 1`] = `
<div>
  <div>
    <h4
      className="sandwishes_name"
    >
      Tikka Naan Roll
    </h4>
    <h5
      className="sandwishes_price"
    >
      Price $
      5
    </h5>
    <button
      className="sandwishes_checkbox"
      id={1}
      onClick={[Function]}
    >
      Add to Basket
    </button>
    <div
      className="sandwishes_extra"
    >
      <div
        className="sandwishes_extra"
      >
        Add Extra: 
        Supplement fromage
         - $
        1
      </div>
      <button
        className="sandwishes_checkbox"
        id={undefined}
        onClick={[Function]}
      >
        Add to Basket
      </button>
    </div>
  </div>
  <div>
    <h4
      className="sandwishes_name"
    >
      Sheek Naan Roll
    </h4>
    <h5
      className="sandwishes_price"
    >
      Price $
      5
    </h5>
    <button
      className="sandwishes_checkbox"
      id={2}
      onClick={[Function]}
    >
      Add to Basket
    </button>
    <div
      className="sandwishes_extra"
    >
      <div
        className="sandwishes_extra"
      >
        Add Extra: 
        Supplement fromage
         - $
        1
      </div>
      <button
        className="sandwishes_checkbox"
        id={undefined}
        onClick={[Function]}
      >
        Add to Basket
      </button>
    </div>
  </div>
  <div>
    <h4
      className="sandwishes_name"
    >
      Pakora Naan Roll
    </h4>
    <h5
      className="sandwishes_price"
    >
      Price $
      5
    </h5>
    <button
      className="sandwishes_checkbox"
      id={3}
      onClick={[Function]}
    >
      Add to Basket
    </button>
    <div
      className="sandwishes_extra"
    >
      <div
        className="sandwishes_extra"
      >
        Add Extra: 
        Supplement fromage
         - $
        1
      </div>
      <button
        className="sandwishes_checkbox"
        id={undefined}
        onClick={[Function]}
      >
        Add to Basket
      </button>
    </div>
  </div>
  <div>
    <h4
      className="sandwishes_name"
    >
      Paneer Naan Roll
    </h4>
    <h5
      className="sandwishes_price"
    >
      Price $
      6.5
    </h5>
    <button
      className="sandwishes_checkbox"
      id={4}
      onClick={[Function]}
    >
      Add to Basket
    </button>
    <div
      className="sandwishes_extra"
    >
      <div
        className="sandwishes_extra"
      >
        Add Extra: 
        Supplement fromage
         - $
        1
      </div>
      <button
        className="sandwishes_checkbox"
        id={undefined}
        onClick={[Function]}
      >
        Add to Basket
      </button>
    </div>
  </div>
  <div>
    <h4
      className="sandwishes_name"
    >
      Frites
    </h4>
    <h5
      className="sandwishes_price"
    >
      Price $
      2.5
    </h5>
    <button
      className="sandwishes_checkbox"
      id={5}
      onClick={[Function]}
    >
      Add to Basket
    </button>
  </div>
  <div>
    <h4
      className="sandwishes_name"
    >
      Pakoras
    </h4>
    <h5
      className="sandwishes_price"
    >
      Price $
      3
    </h5>
    <button
      className="sandwishes_checkbox"
      id={6}
      onClick={[Function]}
    >
      Add to Basket
    </button>
  </div>
</div>
`;
